std/path.d

```diff
@@ -4017,24 +4014,27 @@ string expandTilde(string inputPath) nothrow
                     uint extra_memory_size = 2;
                 else
                     uint extra_memory_size = 5 * 1024;
-                char* extra_memory;
-                scope(exit) free(extra_memory);
+                char[] extra_memory;
 
                 passwd result;
                 while (1)
                 {
-                    extra_memory = cast(char*) realloc(extra_memory, extra_memory_size * char.sizeof);
-                    if (extra_memory == null)
-                        onOutOfMemoryError();
+                    extra_memory ~= new char[extra_memory_size];
```

Comment:
If you change this to

    extra_memory.length += extra_memory_size;

I think the array may be resized in place (i.e., without a complete reallocation) in some cases, whereas an append operation will always reallocate.